From f115c4b4a77763d3693a49e1a49b27efa0896430 Mon Sep 17 00:00:00 2001
From: Daniel Goehring <dgoehrin@os.amperecomputing.com>
Date: Mon, 5 Dec 2022 13:49:28 -0800
Subject: [PATCH 28/31] ac03: target/board: AmpereOne boundary-scan configs [DO
 NOT UPSTREAM]

Add AmpereOne target/board boundary-scan configuration files.

The target configuration file supports silicon and emulation.
The board configuration files support 1 and 2 socket platforms.

Tested on AmpereOne silicon

Change-Id: I355433e09656ba3b2dff4c07353ace7345b9c403
Signed-off-by: Daniel Goehring <dgoehrin@os.amperecomputing.com>
---
 tcl/board/ampere_ac03_1s_bs.cfg | 37 ++++++++++++++++++++
 tcl/board/ampere_ac03_2s_bs.cfg | 40 ++++++++++++++++++++++
 tcl/target/ampere_ac03_bs.cfg   | 60 +++++++++++++++++++++++++++++++++
 3 files changed, 137 insertions(+)
 create mode 100644 tcl/board/ampere_ac03_1s_bs.cfg
 create mode 100644 tcl/board/ampere_ac03_2s_bs.cfg
 create mode 100644 tcl/target/ampere_ac03_bs.cfg

diff --git a/tcl/board/ampere_ac03_1s_bs.cfg b/tcl/board/ampere_ac03_1s_bs.cfg
new file mode 100644
index 000000000..2d1671fb7
--- /dev/null
+++ b/tcl/board/ampere_ac03_1s_bs.cfg
@@ -0,0 +1,37 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+# OpenOCD Board Configuration for the AmpereOne processor
+#
+# Copyright (c) 2022, Ampere Computing LLC
+#
+
+# Argument Description
+#
+# JTAGFREQ
+# Set the JTAG clock frequency
+# Syntax: -c "set JTAGFREQ {freq_in_khz}"
+
+#
+# Configure JTAG speed
+#
+
+if { [info exists JTAGFREQ] } {
+	adapter speed $JTAGFREQ
+} else {
+	adapter speed 100
+}
+
+#
+# Configure Resets
+#
+
+adapter srst delay 100
+jtag_ntrst_delay 100
+reset_config trst_and_srst separate
+
+#
+# Configure Targets
+#
+
+set CHIPNAME s0
+source [find target/ampere_ac03_bs.cfg]
diff --git a/tcl/board/ampere_ac03_2s_bs.cfg b/tcl/board/ampere_ac03_2s_bs.cfg
new file mode 100644
index 000000000..be1012d3e
--- /dev/null
+++ b/tcl/board/ampere_ac03_2s_bs.cfg
@@ -0,0 +1,40 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+# OpenOCD Board Configuration for the AmpereOne processor
+#
+# Copyright (c) 2022, Ampere Computing LLC
+#
+
+# Argument Description
+#
+# JTAGFREQ
+# Set the JTAG clock frequency
+# Syntax: -c "set JTAGFREQ {freq_in_khz}"
+
+#
+# Configure JTAG speed
+#
+
+if { [info exists JTAGFREQ] } {
+	adapter speed $JTAGFREQ
+} else {
+	adapter speed 100
+}
+
+#
+# Configure Resets
+#
+
+adapter srst delay 100
+jtag_ntrst_delay 100
+reset_config trst_and_srst separate
+
+#
+# Configure Targets
+#
+
+set CHIPNAME s1
+source [find target/ampere_ac03_bs.cfg]
+
+set CHIPNAME s0
+source [find target/ampere_ac03_bs.cfg]
diff --git a/tcl/target/ampere_ac03_bs.cfg b/tcl/target/ampere_ac03_bs.cfg
new file mode 100644
index 000000000..875405272
--- /dev/null
+++ b/tcl/target/ampere_ac03_bs.cfg
@@ -0,0 +1,60 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+# OpenOCD Target Configuration for the AmpereOne processor
+#
+# Copyright (c) 2022-2023, Ampere Computing LLC
+#
+
+# Command Line Argument Description
+#
+# CHIPNAME
+# Specifies the chip socket.
+# Will typically be either s0 or s1
+# If not specified, defaults to s0
+# Syntax: -c "set CHIPNAME {s0}"
+#
+# SYSTAPID
+# Can override the SYS TAPID default value if necessary.
+# Experimental Use. Most users will not use this option.
+#
+# DFTTAPID
+# Can override the DFT TAPID default value if necessary.
+# Experimental Use. Most users will not use this option.
+
+#
+# Configure defaults for target
+# These settings can be overridden in the board configuration file
+#
+
+if { [info exists CHIPNAME] } {
+	set _CHIPNAME $CHIPNAME
+} else {
+	set _CHIPNAME s0
+}
+
+if { [info exists SYSTAPID] } {
+	set _SYSTAPID $SYSTAPID
+} else {
+	set _SYSTAPID 0x3BA06477
+}
+
+set DFTTAPID_A0 0x03100A2D
+set DFTTAPID_B0 0x13100A2D
+
+#
+# Definitions
+#
+
+set _SYSTAPNAME sys
+set _DFTTAPNAME dft
+
+#
+# Configure JTAG TAP
+#
+
+if { [info exists DFTTAPID] } {
+	jtag newtap $_CHIPNAME $_DFTTAPNAME.tap -irlen 8 -ircapture 0x1 -irmask 0x3 -expected-id $DFTTAPID
+} else {
+	jtag newtap $_CHIPNAME $_DFTTAPNAME.tap -irlen 8 -ircapture 0x1 -irmask 0x3 -expected-id $DFTTAPID_A0 -expected-id $DFTTAPID_B0
+}
+jtag newtap $_CHIPNAME $_SYSTAPNAME.tap -irlen 4 -ircapture 0x1 -irmask 0x3 -expected-id $_SYSTAPID
-- 
2.25.1

